keps/sig-storage/1472-storage-capacity-tracking/README.md

```diff
@@ -795,79 +840,149 @@ enhancement:
 
+Increased number of pods that are not getting scheduled and apiserver
+load.
+
 * **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?**
 
+No.
+
 * **Is the rollout accompanied by any deprecations and/or removals of features,
   APIs, fields of API types, flags, etc.?**
 
-### Monitoring requirements
+No.
 
-Will be added before the transition to beta.
+### Monitoring requirements
 
 * **How can an operator determine if the feature is in use by workloads?**
 
+The feature is used if all of the following points apply:
+- `CSIDriver.spec.storageCapacity` is true for a CSI driver.
+- There are storage classes for that driver.
+- Volumes are using those storage classes.
+
 * **What are the SLIs (Service Level Indicators) an operator can use to
   determine the health of the service?**
 
+Pod status of the CSI driver deployment, existence of
+`CSIStorageCapacity` objects and ultimately the rate at which volumes
+are being created by the driver. The latter is exposed through the
+existing CSI sidecar metrics support.
+
 * **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
 
+This depends on the workload and CSI driver. A very low rate of
+provisioned volumes per second and few total number of volumes overall
+may be fine for some workloads (long-running applications which use
+specialize storage like PMEM) while others may need a much higher rate
+(adding a local LVM scratch volume to every pod in the system).
+
 * **Are there any missing metrics that would be useful to have to improve
   observability if this feature?**
 
-### Dependencies
+No.
 
-Will be added before the transition to beta.
+### Dependencies
 
 * **Does this feature depend on any specific services running in the cluster?**
 
-### Scalability
+No, just the ones that will also run without it enabled (apiserver,
+kube-scheduler, CSI driver).
 
-Will be added before the transition to beta.
+### Scalability
 
 * **Will enabling / using this feature result in any new API calls?**
 
+Yes.
+
+Enabling it in apiserver and CSI drivers will cause
+`CSIStorageCapacity` objects to be created respectively updated. The
+number of those objects is proportional to the number of storage
+classes and number of distinct storage topology segments. For
+centralized provisioning, the number of segments is probably low. For
+distributed provisioning, the each node where the driver runs
+represents one segment. The rate at which objects depends on how often
+storage usage changes.
+
+Enabling it in kube-scheduler will cause it to cache all
+`CSIStorageCapacity` objects via an informer.
+
 * **Will enabling / using this feature result in introducing new API types?**
 
+Yes, `CSIStorageCapacity`.
+
 * **Will enabling / using this feature result in any new calls to cloud
   provider?**
 
+A CSI driver might have to query the storage backend more often to be
+kept informed about available storage capacity.
```

Comment:
What are the plans? Can we ensure rate-limiting?

In distributed-mode, can we ensure that we won't overloade the provider (we had problems like that in the past).

Reply:
Distributed provisioning is meant for CSI drivers which manage local storage and don't have a backend cloud provider. If they had one, there would be no reason to use distributed provisioning.

Reply:
For central provisioning with a cloud provider we have rate limiting because objects that need to be updated or created go through a rate limiting workqueue.

Reply:
Can you add this here (i.e. distributed is not meant to be used by cloud-provider stuff, and for central we have rate-limiting).

Reply:
Added